### ISSUE TITLE
feat(oxlint): add option to suppress JS plugins experimental warning

### ISSUE THREAD

**Issue by blowery** [oxc-project/oxc#18135]
## Summary

When using JS plugins with oxlint, the following warning is printed to stderr on every run:

```
WARNING: JS plugins are experimental and not subject to semver.
Breaking changes are possible while JS plugins support is under development.
```

This warning is hardcoded in `crates/oxc_linter/src/config/config_builder.rs` and there's no way to suppress it.

## Motivation

While the warning is useful for first-time users to understand the experimental nature of JS plugins, it becomes noise for users who:
- Have acknowledged and accepted the experimental status
- Run oxlint frequently in CI/CD pipelines
- Want clean output for scripting purposes

## Proposed Solution

Add an option to suppress this warning. Possible approaches:

1. **CLI flag**: `--quiet-js-plugin-warning` or include it in existing `--quiet` behavior
2. **Environment variable**: `OXLINT_QUIET_JS_PLUGIN_WARNING=1`
3. **Config option**: Add a field in `.oxlintrc.json` like `"suppressJsPluginWarning": true`

## Workarounds

Currently users can filter stderr, but this is cumbersome and may hide actual errors:
```bash
oxlint 2>&1 | grep -v "WARNING: JS plugins" | grep -v "Breaking changes are possible while JS plugins support is under development"
```

**Comment by connorshea** [comment]
I think, if we do this it should probably be an env var.

**Comment by connorshea** [comment]
This'll be resolved by #18771
